cmd/ascii2der/scanner.go

```diff
@@ -284,6 +284,53 @@ again:
 		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '"' {
 			return s.parseUTF32String()
 		}
+	case 'b':
+		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '`' {
+			s.advance() // Skip the b.
+			s.advance() // Skip the `.
+			bitStr, ok := s.consumeUpTo('`')
+			if !ok {
+				return token{}, &parseError{s.pos, errors.New("unmatched `")}
+			}
+
+			// The leading byte is the number of "extra" bits at the end.
+			var bitCount int
+			var sawPipe bool
+			bits := []byte{0}
+			for i, r := range bitStr {
+				bitsRemaining := (len(bits)-1)*8 - bitCount
+				switch r {
+				case '0', '1':
+					if bitCount%8 == 0 {
+						bits = append(bits, 0)
+					}
+					if r == '1' {
+						bits[len(bits)-1] |= 1 << uint(bitsRemaining)
```

Comment:
I think this still isn't right. :-) You've actually ordered the bits by: `1<<0`, `1<<7`, `1<<6`, `1<<5`, `1<<4`, `1<<3`, `1<<2`, `1<<1`. It's hard to tell because all your tests alternate bits. :-) Maybe we should add some very basic tests <code>b\`1\`</code>, <code>b\`11\`</code>, through, <code>b\`1111111\`</code>. I find that a lot easier to tell what's going on in terms of bit ordering. (Maybe something to be said for some long-ish random input too?)

Part of what's going on is that `bitsRemaining` is computed before `bitCount` and the other part is it should be `bitsRemaining - 1`. But perhaps it'd be easier just to write:
```
   if bitCount%8 == 0 {
     bits = append(bits, 0)
   }
   if r == '1' {
     // Or maybe we keep the byte index at len(bits)-1? Don't feel very strongly... pairing /8 with
     // %8 just feels tidy.
     bits[bitCount/8 + 1] |= 1 << uint(7 - bitCount%8)
   }
```

Then we can move `bitsRemaining` into the `|` case.

Reply:
That was surprisingly fussy. Big endian bits is... quite unintuitive.